File: src/main/java/com/yauhenl/neuro/Food.java
package com.yauhenl.neuro;

import processing.core.PVector;

class Food {
    private final PVector location;
    private Environment env;

    PVector getLocation() {
        return location;
    }

    Food(Environment environment) {
        env = environment;
        location = new PVector(env.random(env.width), env.random(env.height));
    }

    void display() {
        env.noStroke();
        env.fill(200, 50, 0);
        env.rect(location.x, location.y, 8, 8);
    }
}

File: src/main/kotlin/com/yauhenl/kotlin/neuro/Food.kt
package com.yauhenl.kotlin.neuro

import com.yauhenl.kotlin.neuro.Environment
import processing.core.PVector

class Food internal constructor(private val env: Environment) {
    val location: PVector = PVector(env.random(env.width.toFloat()), env.random(env.height.toFloat()))

    fun display() {
        env.noStroke()
        env.fill(200f, 50f, 0f)
        env.rect(location.x, location.y, 8f, 8f)
    }
}
